docs/source/snapshot.rst

```diff
@@ -0,0 +1,104 @@
+Snapshot
+========
+
+:class:`~torchsnapshot.Snapshot` is the core API of TorchSnapshot. The class represents application state persisted in storage. A user can take a snapshot of an application via :meth:`Snapshot.take() <torchsnapshot.Snapshot.take>` (a.k.a saving a checkpoint), or restore the state of an application from a snapshot via :meth:`Snapshot.restore() <torchsnapshot.Snapshot.restore>` (a.k.a loading a checkpoint).
+
+
+Describing the application state
+--------------------------------
+
+Before using :class:`~torchsnapshot.Snapshot` to save or restore application state, the user needs to **describe the application state**. This is done by creating a dictionary that contains all **stateful objects** that the user wishes to capture as application state:
```

Comment:
nit: mention that the code block below is an example

Reply:
Was this not obvious? (not asking rhetorically)

Reply:
On second thought this is fine, I'm used to examples starting with `>>>`.